aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -1000,10 +1006,10 @@ Tensor& intersection_binary_op_sparse_dense_out(
 
     return indices;
   }();
-
-  get_sparse_impl(res)->raw_resize_(res_sparse_dim, res_dense_dim, res_shape);
-  get_sparse_impl(res)->set_indices_and_values_unsafe(res_indices, res_values);
-  get_sparse_impl(res)->set_nnz_and_narrow(res_nnz);
+  auto* res_impl = get_sparse_impl(res);
+  res_impl->raw_resize_(res_sparse_dim, res_dense_dim, res_shape);
+  res_impl->set_indices_and_values_unsafe(res_indices, res_values);
+  res_impl->set_nnz_and_narrow(res_nnz);
```

Comment:
I see these 4 lines + setting `_coalesced_` repeating in a few places, could we refactor this into a utility function?

Reply:
Such a utility function would be equivalent to `zeros_like_` (in place `zeros_like`).

It would make sense to tackle this as a follow-up because there will be many unrelated changes for this fix.